File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/json/OkHttpClientUtil.java
package com.elveselimoski.starterapp.json;

import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import org.jetbrains.annotations.NotNull;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

public final class OkHttpClientUtil {

    private static final long TIMEOUT_IN_SECONDS = 5L;
    private static OkHttpClient.Builder defaultHttpCliendBuilder = new OkHttpClient.Builder()
            .readTimeout(TIMEOUT_IN_SECONDS, TimeUnit.SECONDS)
            .connectTimeout(TIMEOUT_IN_SECONDS, TimeUnit.SECONDS);
    public static OkHttpClient defaultHttpClient = defaultHttpCliendBuilder.build();
//        .addInterceptor(new ExampleInterceptor());

    private OkHttpClientUtil() {
    }

    class ExampleInterceptor implements Interceptor {
        @NotNull
        @Override
        public Response intercept(@NotNull Chain chain) throws IOException {
            Request request = chain.request();
            request = request.newBuilder()
//                    .addHeader("header-name", StarterApp.getInstance().getSomeHeader())
                    .build();
            return chain.proceed(request);
        }
    }

}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/json/OkHttpClientUtil.kt
package com.elveselimoski.starterapp.json

import androidx.annotation.NonNull
import com.elveselimoski.starterapp.StarterApp
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.Response
import java.util.concurrent.TimeUnit

object OkHttpClientUtil {
    private const val TIMEOUT_IN_SECONDS = 5L

    private val defaultHttpCliendBuilder: OkHttpClient.Builder = OkHttpClient.Builder()
        .readTimeout(TIMEOUT_IN_SECONDS, TimeUnit.SECONDS)
        .connectTimeout(TIMEOUT_IN_SECONDS, TimeUnit.SECONDS)
//        .addInterceptor(ExampleInterceptor())

    val defaultHttpClient: OkHttpClient = defaultHttpCliendBuilder.build()

    class ExampleInterceptor : Interceptor {
        override fun intercept(chain: Interceptor.Chain): Response {
            var request = chain.request()
            request = request.newBuilder()
//                .header("header-name", StarterApp.getInstance().getSomeHeader())
                .build()
            return chain.proceed(request)
        }

    }

}